conans/client/importer.py

```diff
@@ -143,7 +143,7 @@ def __call__(self, pattern, dst="", src="", root_package=None, folder=False,
         matching_paths = self._get_folders(root_package)
         for name, matching_path in matching_paths.items():
             final_dst_path = os.path.join(real_dst_folder, name) if folder else real_dst_folder
-            file_copier = FileCopier(matching_path, final_dst_path)
+            file_copier = FileCopier(matching_path, final_dst_path, make_writable=True)
```

Comment:
I  think I wouldn't delegate this to the copier.

As you get the list of copied files, in this ``imports`` function, I would check the environment variable that defines that the cache is in "READ_ONLY" mode, and if that is the case, I would add write permissions to all files included in the list of copied files.

Reply:
I will move the modification of permissions then.

As to doing that only when the cache is read-only: wouldn't that be inconsistent? Suppose you have a package with a read-only file. Then when the cache is in normal mode, the file would be deployed and imported as read-only. When you then set the cache in read-only mode, that file would suddenly be deployed and imported as read-write.

Reply:
But that is the expected behavior. If one package has some file in read-only, it is their responsibility and decission. Conan shouldn't change that.

So this is why I said that I was not completely happy with this feature, there is no perfect solution. In those cases, I prefer to opt for the least changing approach. Only users with the env-var defined will have conan changing R-W permissions of files from the cache.

Reply:
Well, I thought of it as follows: when a file is copied from the Conan cache to user-space (through an import or deploy), Conan is no longer the owner of or responsible for the copy. The user should have write access. A package maintainer should not have be able to control how their packages are used. Only how they are shipped.

But I agree that in this case, where there's no solution that pleases all, changing as little as possible is probably best.

I'll rework my solution and try to add some tests.